postgres/osmap.yaml

```diff
@@ -91,9 +91,43 @@ RedHat:
 {% endif %}
 
 Suse:
+  pkg_repo:
+    name: pgdg-sles-{{ release }}
+    humanname: PostgreSQL {{ repo.version }} $releasever - $basearch
+    #Using sles-12 upstream repo for opensuse
+    baseurl: 'https://download.postgresql.org/pub/repos/zypp/{{ repo.version }}/suse/sles-12-$basearch'
+    key_url: 'https://download.postgresql.org/pub/repos/zypp/{{ repo.version }}/suse/sles-12-$basearch/repodata/repomd.xml.key'
+    gpgcheck: 1
+    gpgautoimport: True
+
+{% if repo.use_upstream_repo %}
+  {# Pillars needed are 'use_upstream_repo: True' and 'version: n.n'. #}
+  {# Avoid setting package names as pillars, as may corrupt postgres. #}
+  {% set lib_dir = '/var/lib/pgsql/' ~ repo.version ~ '/data' %}
+
+  pkg: postgresql{{ release }}-server
+  pkg_client: postgresql{{ release }}
+  conf_dir: {{ lib_dir }}
+  service: postgresql-{{ repo.version }}
+
+  #This is postgresql-libs in defaults.yml but upstream is libpqxx
+  pkg_libpq_dev: libpqxx
+  pkg_dev: postgresql{{ release }}-devel
+  pkg_libs: postgresql{{ release }}-libs
+
+  prepare_cluster:
+    #Full path needed as initdb is NOT 'cross version compatible' binary
+    command: /usr/pgsql-{{ repo.version }}/bin/initdb --pgdata='{{ lib_dir }}'
```

Comment:
This is generally OK. But I would recommend you to consider providing `bin_dir` and `*_bins` parameters similar to RedHat section for `postgres.manage` state to work on Suse out of the box as well.
But that could be added later.

Reply:
https://github.com/saltstack-formulas/postgres-formula/issues/189

Reply:
Also https://github.com/saltstack-formulas/postgres-formula/issues/190 captures problem regarding package names in pillar data.